waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -773,17 +773,20 @@ proc mountRlnRelayDynamic*(node: WakuNode,
   var 
     keyPair: MembershipKeyPair
     rlnIndex: MembershipIndex
-  if memKeyPair.isNone: # if non provided, create one and register to the contract
-    trace "no rln-relay key is provided, generating one"
-    let keyPairOpt = rln.membershipKeyGen()
-    doAssert(keyPairOpt.isSome)
-    keyPair = keyPairOpt.get()
-    # register the rln-relay peer to the membership contract
-    let regIndexRes = await  register(idComm = keyPair.idCommitment, ethAccountAddress = ethAccAddr, ethAccountPrivKey = ethAccountPrivKey, ethClientAddress = ethClientAddr, membershipContractAddress = memContractAddr)
-    # check whether registration is done
-    doAssert(regIndexRes.isOk())
-    rlnIndex = regIndexRes.value
-    debug "peer is successfully registered into the membership contract"
+  if memKeyPair.isNone: 
+    if ethAccountPrivKeyOpt.isSome: # if non provided, create one and register to the contract
```

Comment:
The comment is a bit misleading here and should refer to memKeyPair, although its non-existence doesn't automatically imply registration since you now check if an ethAccountPrivKey is set

Reply:
Thanks, clarified in [1945edc](https://github.com/status-im/nwaku/pull/1051/commits/1945edc35df26cc336a73a485e20be8549f0afad)